go/vt/vtgate/executor.go

```diff
@@ -229,7 +229,7 @@ func (e *Executor) execute(ctx context.Context, safeSession *SafeSession, sql st
 	case sqlparser.StmtRollback:
 		return e.handleRollback(ctx, safeSession, sql, bindVars, logStats)
 	case sqlparser.StmtSet:
-		return e.handleSet(ctx, safeSession, sql, bindVars, logStats)
+		return e.handleSet(ctx, safeSession, sql, bindVars, destKeyspace, logStats)
```

Comment:
I think automagically picking up the keyspace from the current target is a bit surprising. Existing uses of the `SET` statement aren't contextual like that.

Reply:
Done